src/main/java/com/leveluptasks/controller/UserController.java

```diff
@@ -61,6 +62,17 @@ public ResponseEntity<User> login(@RequestBody UserDto userDto) throws NoSuchAlg
         }
     }
 
+    @Operation(summary = "Find user with email", description = "Find user with email")
+    @PostMapping("/email/{email}")
+    public ResponseEntity<User> findUserByEmail(@PathVariable String email) throws NoSuchAlgorithmException {
+        Optional<User> user = userService.findByEmail(email);
+        if (user.isPresent()) {
+            return ResponseEntity.ok(user.get());
```

Comment:
you can remove the whole optional part and add it in the service like in the controller you just return the user